### PR TITLE
[AV-1736] Move network runner logs and configs from tmp to .avalanche-cli

### PR DESCRIPTION
**Pull request by dasconnor**
Place network runner logs in the `.avalanche-cli/runs` directory.

Requires a network runner update before merging: https://github.com/ava-labs/avalanche-network-runner/pull/167

### CODE REVIEW

File: pkg/app/app.go [hunk 1]
```diff
@@ -17,6 +18,8 @@ import (
 
 const (
 	WriteReadReadPerms = 0o644
+
+	runDir = "runs"
```
Comment (felipemadero) [inline pkg/app/app.go:22]: probably shall go to constants.go (same as bin, and any other dirname for the tree under .avalanche-cli)

File: pkg/binutils/processes.go [hunk 2]
```diff
@@ -111,12 +114,18 @@ func GetServerPID() (int, error) {
...
-func StartServerProcess(log logging.Logger) error {
+func StartServerProcess(app app.Avalanche) error {
 	thisBin := reexec.Self()
 
 	args := []string{"backend", "start"}
 	cmd := exec.Command(thisBin, args...)
-	outputFile, err := os.CreateTemp("", "avalanche-cli-backend*")
+
+	outputFilePrefix := path.Join(app.GetRunDir(), "server")
+	outputFilePath, err := utils.MkDirWithTimestamp(outputFilePrefix)
```
Comment (felipemadero) [inline pkg/binutils/processes.go:124]: is this a file or a dir? names outputFilePrefix, etc... may be confusing

File: pkg/binutils/processes.go [hunk 3]
```diff
@@ -111,12 +114,18 @@ func GetServerPID() (int, error) {
...
 // it just executes `avalanche-cli backend start`
-func StartServerProcess(log logging.Logger) error {
+func StartServerProcess(app app.Avalanche) error {
 	thisBin := reexec.Self()
 
 	args := []string{"backend", "start"}
 	cmd := exec.Command(thisBin, args...)
-	outputFile, err := os.CreateTemp("", "avalanche-cli-backend*")
+
+	outputFilePrefix := path.Join(app.GetRunDir(), "server")
```
Comment (felipemadero) [inline pkg/binutils/processes.go:123]: why a dir with tmp for each execution? maybe only a dir like server-info, and inside it the different execution files with timestamps?

File: pkg/binutils/processes.go [hunk 4]
```diff
@@ -141,7 +150,7 @@ func StartServerProcess(log logging.Logger) error {
 	}
 	err = os.WriteFile(constants.ServerRunFile, rfBytes, perms.ReadWrite)
```
Comment (felipemadero) [inline pkg/binutils/processes.go:151]: probably we want to move ServerRunFile to .avalanche-cli?

File: pkg/app/app.go (hunk 1 above)
Comment (dasconnor) [reply pkg/app/app.go]: done

File: pkg/binutils/processes.go (hunk 2 above)
Comment (dasconnor) [reply pkg/binutils/processes.go]: renamed

File: pkg/binutils/processes.go (hunk 4 above)
Comment (dasconnor) [reply pkg/binutils/processes.go]: good catch, fixed

File: cmd/start.go [hunk 5]
```diff
@@ -42,7 +45,14 @@ func startNetwork(cmd *cobra.Command, args []string) error {
 	ctx := binutils.GetAsyncContext()
 
 	ux.Logger.PrintToUser(startMsg)
-	_, err = cli.LoadSnapshot(ctx, snapshotName)
+
+	outputDirPrefix := path.Join(app.GetRunDir(), "restart")
```
Comment (felipemadero) [inline cmd/start.go:49]: shall also use the current rundir of the server? as the deploy

File: pkg/binutils/processes.go [hunk 6]
```diff
@@ -26,6 +29,12 @@ import (
 // gRPCTimeout is a common error message if the gRPC server can't be reached
 var gRPCTimeout = errors.New("Timed out trying to contact backend controller, it is most probably not running.")
 
+var latestRunDir string
+
+func GetLatestRunDir() string {
```
Comment (felipemadero) [inline pkg/binutils/processes.go:34]: is it going to be available in a subsecuent deploy or start command? because the server will continue to be alive.
It seems the idea of putting the ANR server logs in the same directory as the ANR local network logs is ok for debugging purposes. 
But as the server lifetime exceeds that of a ANR local network (you can deploy, then stop, then deploy, stop, start, all using the same server).
The directory struct seems to be like this:
```
server directory
- ANR directory 1
- ...
- ANR directoryN
```
What do you think on this?

Comment (felipemadero) [reply pkg/binutils/processes.go:34]: Anyway GetLatestRunDir maybe can be renamed to something related to backend or server rundir, and also, needs to be persistent in some way

File: cmd/start.go (hunk 5 above)
Comment (dasconnor) [reply cmd/start.go:49]: since it's restarting, it gets a new folder

File: pkg/binutils/processes.go (hunk 6 above)
Comment (dasconnor) [reply pkg/binutils/processes.go:34]: Yeah, it works for the current structure but when we get subsequent deploys working, we might want to tweak things a little. It's hard to test right now though. We can actually grab the latest dir persistently by getting the latest timestamped directory.

File: pkg/subnet/local_test.go [hunk 7]
```diff
@@ -164,7 +169,7 @@ func TestGetLatestAvagoVersion(t *testing.T) {
 func getTestClientFunc() (client.Client, error) {
 	c := &mocks.Client{}
 	fakeStartResponse := &rpcpb.StartResponse{}
-	c.On("Start", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(fakeStartResponse, nil)
+	c.On("Start", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(fakeStartResponse, nil)
```
Comment (holisticode) [inline pkg/subnet/local_test.go:172]: Huh, why is this needed?

Comment (dasconnor) [reply pkg/subnet/local_test.go:172]: A test was failing, not sure why it didn't break the pipeline on main. This was required for it to pass.

**Review by holisticode — APPROVED** [review]

File: pkg/subnet/local_test.go (hunk 7 above)
Comment (dasconnor) [reply pkg/subnet/local_test.go:172]: Looks like I added a new argument to start, h/t @holisticode
